db.go

```diff
@@ -1554,6 +1558,76 @@ func (db *DB) EnforceSnapshotRetention(ctx context.Context, timestamp time.Time)
 	return minSnapshotTXID, nil
 }
 
+// EnforceL0RetentionByTime retains L0 files until they have been compacted into
+// L1 and have existed for at least L0Retention.
+func (db *DB) EnforceL0RetentionByTime(ctx context.Context) error {
+	if db.L0Retention <= 0 {
+		return nil
+	}
+
+	db.Logger.Debug("enforcing l0 retention", "retention", db.L0Retention)
+
+	// Determine the highest TXID that has been compacted into L1.
+	itr, err := db.Replica.Client.LTXFiles(ctx, 1, 0, false)
+	if err != nil {
+		return fmt.Errorf("fetch l1 files: %w", err)
+	}
+	var maxL1TXID ltx.TXID
+	for itr.Next() {
+		info := itr.Item()
+		if info.MaxTXID > maxL1TXID {
+			maxL1TXID = info.MaxTXID
+		}
+	}
+	if err := itr.Close(); err != nil {
+		return fmt.Errorf("close l1 iterator: %w", err)
+	}
+	if maxL1TXID == 0 {
+		return nil
+	}
+
+	threshold := time.Now().Add(-db.L0Retention)
+	itr, err = db.Replica.Client.LTXFiles(ctx, 0, 0, false)
+	if err != nil {
+		return fmt.Errorf("fetch l0 files: %w", err)
+	}
+	defer itr.Close()
+
+	var deleted []*ltx.FileInfo
+	var lastInfo *ltx.FileInfo
+	for itr.Next() {
+		info := itr.Item()
+		lastInfo = info
+
+		if info.MaxTXID <= maxL1TXID && !info.CreatedAt.IsZero() && !info.CreatedAt.After(threshold) {
+			deleted = append(deleted, info)
+		}
```

Comment:
Do we have instances where `info.CreatedAt` would be zero? If so, those files would never be deleted.